onnxruntime/core/platform/posix/env.cc

```diff
@@ -70,6 +73,18 @@ struct FileDescriptorTraits {
   }
 };
 
+std::string where_am_i(){
+    Dl_info info;
+    if (dladdr((void*)&where_am_i, &info) != 0){
+        char path[PATH_MAX];
+        strcpy(path, info.dli_fname);
```

Comment:
please use strncpy or strlcpy.

Reply:
I think it doesn't work the executable files. For example: onnx_test_runner, onnxruntime_perf_test. Usually they will be installed to locations like /usr/bin  or /usr/local/bin. You won't be able to find the shared libraries in the same directory. 
Anyway, something is better than nothing.

Reply:
if that's the case, how does our shared library based EP works on windows? it will have the same issue, right?

Reply:
Windows programs usually have a different layout. 
1. Windows doesn't have bin/lib dir seperation. Usually DLLs and Exes are in the same dir.
2. Linux usually put all shared libraries in a centralized place(/usr/lib, /usr/local/lib, ...) and managed by ldconfig. Windows doesn't have such things
3. Python is different.  We hit this error because a tool from manylinux docker image erased the rpath setting we put in.  But manylinux is only for Linux.

Reply:
we could modify our package script to add the path back after audit tool right? 
the title of this PR is a little misleading. there's nothing inherently wrong with CentOS. It's just our tooling for creating a python package reset some elf headers we wrote to the binary.